python/pyarrow/tests/test_parquet.py

```diff
@@ -641,6 +641,36 @@ def make_sample_file(table_or_df):
     return pq.ParquetFile(buf)
 
 
+def test_byte_stream_split():
+    # This is only a smoke test.
+    arr_float = pa.array(list(map(float, range(100))))
+    data_float = [arr_float, arr_float]
+    table = pa.Table.from_arrays(data_float, names=['a', 'b'])
+
+    # Check with byte_stream_split for both columns.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=False, use_byte_stream_split=True)
+
+    # Check with byte_stream_split for column 'b' and dictionary
+    # for column 'a'.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=['a'],
+                     use_byte_stream_split=['b'])
+
+    # Check with a collision for both columns.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=['a', 'b'],
+                     use_byte_stream_split=['a', 'b'])
+
+    # Try to use the wrong data type with the byte_stream_split encoding.
+    # This should throw an exception.
+    arr_int = pa.array(list(map(int, range(10))))
+    table = pa.Table.from_arrays([arr_int], names=['tmp'])
+    with pytest.raises(IOError):
```

Comment:
Why `IOError`? I would expect `ValueError`, since it's a parameter error.

Reply:
This gets called https://github.com/apache/arrow/blob/master/cpp/src/parquet/arrow/writer.cc#L481
Parquet throws a ParquetException, Arrow captures it and returns an error.
From the pyarrow check_status gets called which raises an IOError exception.

Parquet also doesn't seem to expose anything but ParquetException and Arrow always consider failures to write a file as an IOError. Of course, we could do such validation on the pyarrow side but this doesn't really help with the Arrow interface.

Do you think it's reasonable to leave it as IOError?

Reply:
Well, if Parquet doesn't make a difference, then we'll have to live it I guess :-) Too bad.